wgpu-core/src/swap_chain.rs

```diff
@@ -135,47 +139,48 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             ),
         };
 
-        let view_id = image.map(|image| {
-            let view = resource::TextureView {
-                inner: resource::TextureViewInner::SwapChain {
-                    image,
-                    source_id: Stored {
-                        value: swap_chain_id,
-                        ref_count: sc.life_guard.add_ref(),
+        let view_id = image
```

Comment:
could we write this as a `match` now? Passing a closure now seems weird with the `Result` and `transpose` at the end